### PR TITLE
tests: multiple fixes

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

With #17721 I ran tests on multiple platforms and discovered failing tests that were unrelated to that PR. This PR fixes them. The affected platforms are `avr` and `z1`, running the affected tests on those platforms should be enough.

### Testing procedure

- Green Murdock
- I'll post run-results

### CODE REVIEW

File: tests/sys_sema_inv/main.c [hunk 1]
```diff
@@ -43,7 +43,6 @@ static void *thread_count(void *arg)
     if (sema_inv_post(ctx->sync)) {
         printf("THREAD %u woke main thread\n", ctx->id);
     }
-
```
Comment (kaspar030) [inline tests/sys_sema_inv/main.c:46]: don't remove newline here?

File: tests/sys_sema_inv/main.c [hunk 2]
```diff
@@ -58,7 +57,6 @@ static void *thread_bit(void *arg)
     if (sema_inv_post_mask(ctx->sync, 1 << ctx->id)) {
         printf("THREAD %u woke main thread\n", ctx->id);
     }
-
```
Comment (kaspar030) [inline tests/sys_sema_inv/main.c:61]: don't remove?

**Comment by kaspar030** [comment]
would you split out the z1 socat fixes (88a8b3622bd2cb0eb984bb9f28d12de6eb7f17d2)? (maybe we find a better solution that doesn't require board specific test fixes)

**Comment by fjmolinas** [comment]
@kaspar030 done and split out

**Review by kaspar030 — APPROVED** [review]
ACK.

**Comment by fjmolinas** [comment]
Thanks!

**Pull request merged**